api/dataprocessor.go

```diff
@@ -188,6 +188,8 @@ func (s *Server) getTargets(reqs []models.Req) ([]models.Series, error) {
 	if len(localReqs) > 0 {
 		wg.Add(1)
 		go func() {
+			// the only errors returned are from us catching panics, so we should treat them
+			// all as internalServerErrors
```

Comment:
there are no "http things" (not sure what that actually means) in dataprocessor.

The comment just conveys that we dont need to "tag" the errors returned to the caller with a statusCode, by using response.NewError(code, msg) or similar, as they should all be treated as internalServer errors.